src/Http/Http.Features/src/IHttpResponseBodyFeature.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using System.IO.Pipelines;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Http.Features
+{
+    /// <summary>
+    /// An aggregate of the different ways to interact with the response body.
+    /// </summary>
+    public interface IHttpResponseBodyFeature
+    {
+        /// <summary>
+        /// The <see cref="System.IO.Stream"/> for writing the response body.
+        /// </summary>
+        Stream Stream { get; }
+
+        /// <summary>
+        /// A <see cref="PipeWriter"/> representing the response body, if any.
+        /// </summary>
+        PipeWriter Writer { get; }
+
+        /// <summary>
+        /// Opts out of write buffering for the response.
+        /// </summary>
+        void DisableBuffering();
+
+        /// <summary>
+        /// Starts the response by calling OnStarting() and making headers unmodifiable.
```

Comment:
nit: Don't mention that it calls OnStarting() on an interface. It's up to the implementer to decide what to do.

Reply:
I feel like that's the contract. At a minimum, calling StartAsync should induce OnStarting and commit the headers.

Reply:
That's certainly the intent anyways, but things like buffering could prevent that.